test/view/iota.cpp

```diff
@@ -34,20 +35,40 @@ struct Int
 
 template<typename Integral>
 void test_iota_minus() {
-  using namespace ranges;
-  using D = detail::iota_difference_t<Integral>;
-  using I = Integral;
-  Integral max = std::numeric_limits<Integral>::max();
-
-  CHECK(detail::iota_minus_(I(0), I(0)) == D(0));
-  CHECK(detail::iota_minus_(I(0), I(1)) == D(-1));
-  CHECK(detail::iota_minus_(I(1), I(0)) ==  D(1));
-  CHECK(detail::iota_minus_(I(1), I(1)) == D(0));
-
-  CHECK(detail::iota_minus_(I(max - I(1)), I(max - I(1))) == D(0));
-  CHECK(detail::iota_minus_(I(max - I(1)), I(max)) == D(-1));
-  CHECK(detail::iota_minus_(I(max), I(max - I(1))) == D(1));
-  CHECK(detail::iota_minus_(I(max), I(max)) == D(0));
+    using namespace ranges;
+    using D = detail::iota_difference_t<Integral>;
+    using I = Integral;
+    Integral max = std::numeric_limits<Integral>::max();
+
+    CHECK(detail::iota_minus_(I(0), I(0)) == D(0));
+    CHECK(detail::iota_minus_(I(0), I(1)) == D(-1));
+    CHECK(detail::iota_minus_(I(1), I(0)) ==  D(1));
+    CHECK(detail::iota_minus_(I(1), I(1)) == D(0));
+
+    CHECK(detail::iota_minus_(I(max - I(1)), I(max - I(1))) == D(0));
+    CHECK(detail::iota_minus_(I(max - I(1)), I(max)) == D(-1));
+    CHECK(detail::iota_minus_(I(max), I(max - I(1))) == D(1));
+    CHECK(detail::iota_minus_(I(max), I(max)) == D(0));
+}
+
+template <typename U>
+void test_wrap_around() {
+    using namespace ranges;
+    auto max = std::numeric_limits<U>::max();
```

Comment:
I would get rid of `max`, and use `U(-1)`, `U(-2)`, and `U(-3)` in place of `max`, `U(max - 1)`, and `U(max - 2)`. This function then behaves identically for both signed and unsigned types.